src/Storages/StorageFile.cpp

```diff
@@ -170,23 +116,23 @@ void listFilesWithRegexpMatchingImpl(
     std::vector<std::string> & result,
     bool recursive)
 {
-    /// regexp for {expr1,expr2,expr3} or {M..N}, where M and N - non-negative integers, expr's should be without "{", "}", "*" and ","
-    static const re2::RE2 enum_or_range(R"({([\d]+\.\.[\d]+|[^{}*,]+,[^{}*]*[^{}*,])})");
+    const size_t first_glob_pos = for_match.find_first_of("*?{");
 
-    std::string_view for_match_view(for_match);
-    std::string_view matched;
-    if (RE2::FindAndConsume(&for_match_view, enum_or_range, &matched))
+    if (first_glob_pos == std::string::npos)
     {
-        std::string buffer(matched);
-        if (buffer.find(',') != std::string::npos)
+        try
         {
-            expandSelector(path_for_ls, for_match, total_bytes_to_read, result, recursive);
-            return;
+            fs::path path = fs::canonical(path_for_ls + for_match);
```

Comment:
@zvonand shouldn't it be `fs::absolute` here? Otherwise we cannot have symbolic link in `user_files`.

Reply:
Hm, maybe you're right and I missed this one